deploy/charts/harvester/templates/job.yaml

```diff
@@ -0,0 +1,82 @@
+kind: Job
+metadata:
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+  # NB(thxCode): name should not be customized as below:
+  # name: {{ template "harvester.fullname" . }}-pre-delete
+  # because we can easily confirm this resource from the corresponding namespace.
+  name: harvester-pre-delete
+  labels:
+{{ include "harvester.labels" . | indent 4 }}
+    app.kubernetes.io/name: pre-delete
+    app.kubernetes.io/component: job
+spec:
+{{- if .Values.jobs.preDelete.activeDeadlineSeconds }}
+  activeDeadlineSeconds: {{ .Values.jobs.preDelete.activeDeadlineSeconds }}
+{{- end }}
+{{- if .Values.jobs.preDelete.backoffLimit }}
+  backoffLimit: {{ .Values.jobs.preDelete.backoffLimit }}
+{{- end }}
+{{- if .Values.jobs.preDelete.ttlSecondsAfterFinished }}
+  ttlSecondsAfterFinished: {{ .Values.jobs.preDelete.ttlSecondsAfterFinished }}
+{{- end }}
+  template:
+    metadata:
+      name: harvester-pre-delete
+      labels:
+{{ include "harvester.immutableLabels" . | indent 8 }}
+        app.kubernetes.io/name: pre-delete
+        app.kubernetes.io/component: job
+    spec:
+      # NB(thxCode): reuse the "harvester" ServiceAccount under the same namespace.
+      serviceAccountName: harvester
+{{- if .Values.jobs.preDelete.restartPolicy }}
+      restartPolicy: {{ .Values.jobs.preDelete.restartPolicy }}
+{{- end }}
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: beta.kubernetes.io/os
+                    operator: In
+                    values:
+                      - linux
+              - matchExpressions:
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - linux
+{{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
+      containers:
+        - name: pre-delete
+          image: {{ .Values.jobs.preDelete.containers.kubectl.image.repository }}:{{ .Values.jobs.preDelete.containers.kubectl.image.tag }}
+          imagePullPolicy: {{ .Values.jobs.preDelete.containers.kubectl.image.imagePullPolicy }}
+          command:
+            - "/bin/bash"
+          args:
+            - "-c"
+{{- if and (index .Values "kubevirt-operator" "enabled") (index .Values "cdi-operator" "enabled") }}
+            - "echo '[INFO] deleting cdi ...'; kubectl delete cdi --wait --all; echo '[INFO] deleting kubevirt ...'; kubectl delete kubevirt --wait --all; exit"
+{{- else if index .Values "kubevirt-operator" "enabled" }}
+            - "echo '[INFO] deleting kubevirt ...'; kubectl delete kubevirt --wait --all; exit"
+{{- else }}
+            - "echo '[INFO] deleting cdi ...'; kubectl delete cdi --wait --all; exit"
+{{- end }}
```

Comment:
Looks two conditions are sufficient for this
```
if(kvo enabled) delete; endif; 
if(cdio enabled) delete; endif;
```

Reply:
Yes, in fact, they are only two conditions. However, here needs to write in one line behind `/bin/bash -c`. I have merged these conditions in one line.